app/src/main/java/io/gnosis/safe/ui/transaction/TransactionViewHolderFactory.kt

```diff
@@ -0,0 +1,76 @@
+package io.gnosis.safe.ui.transaction
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.viewbinding.ViewBinding
+import io.gnosis.data.models.Transaction
+import io.gnosis.safe.databinding.EntryTransactionCancelledBinding
+import io.gnosis.safe.databinding.EntryTransactionFailedBinding
+import io.gnosis.safe.databinding.EntryTransactionPendingBinding
+import io.gnosis.safe.databinding.EntryTransactionSuccessBinding
+import io.gnosis.safe.ui.adapter.BaseFactory
+import io.gnosis.safe.ui.adapter.BaseViewHolder
+import io.gnosis.safe.ui.adapter.UnsupportedViewType
+
+enum class TransactionViewType {
+    PENDING, SUCCESS, CANCELLED, FAILED
```

Comment:
Should we have different ViewHolders based on the state (PENDING, SUCCESS, CANCELLED, FAILED) of the transaction or the type (custom_transaction, settings_change, transfer, creation_call(?)) of the transaction?

Reply:
I would say type

Reply:
our tx layout segregation is also based on type

Reply:
This is an outdated change.